workflow/lib/preprocess/preprocess.py

```diff
@@ -2,94 +2,111 @@
+                # Add more metadata fields
+                metadata.update({
+                    "channels": images.channel_count(),
+                    "pixel_size_x": getattr(images.metadata, 'pixel_size_x', None),
+                    "pixel_size_y": getattr(images.metadata, 'pixel_size_y', None),
+                    "z_step": getattr(images.metadata, 'z_step', None),
+                })
+
+                df = pd.DataFrame(metadata)
+
+                # Sample z-planes if interval is specified and z_data exists
+                if z_interval and len(metadata["z_data"]) > 0:
+                    df = df.iloc[::z_interval, :]
+                
+                if verbose:
+                    print(f"Found {len(df)} positions for tile {tile}")
+                
+                all_metadata.append(df)
+                
+        except Exception as e:
+            print(f"Error processing {file_path}: {str(e)}")
+            continue
 
-        with ND2Reader(file_path) as images:
-            raw_metadata = images.parser._raw_metadata
-
-            data = {
-                "x_data": raw_metadata.x_data,
-                "y_data": raw_metadata.y_data,
-                "z_data": images.metadata["z_coordinates"],
-                "pfs_offset": raw_metadata.pfs_offset,
-                "field_of_view": tile,
-                "filename": file_path,
-            }
-
-            df = pd.DataFrame(data)
-
-            if "z_levels" in images.metadata and set(
-                images.metadata["z_levels"]
-            ) == set(range(0, 4)):
-                df = df.iloc[::4, :]
-            all_metadata.append(df)
-
-    # Combine all metadata
     if all_metadata:
         combined_metadata = pd.concat(all_metadata, ignore_index=True)
-
-        # Convert 'field_of_view' to numeric, coercing any non-numeric values to NaN
         combined_metadata["field_of_view"] = pd.to_numeric(
             combined_metadata["field_of_view"], errors="coerce"
         )
-        # Sort by 'field_of_view' in ascending order
-        combined_metadata = combined_metadata.sort_values("field_of_view").reset_index(
-            drop=True
-        )
-        return combined_metadata
-    else:
-        print("No valid ND2 files found in the provided list.")
-        return pd.DataFrame()  # Return an empty DataFrame if no files were processed
+        return combined_metadata.sort_values("field_of_view").reset_index(drop=True)
+    
+    print("No valid ND2 files found in the provided list.")
+    return pd.DataFrame()
 
 
-def nd2_to_tiff(file: str, channel_order_flip: bool = False) -> np.ndarray:
-    """Converts a single ND2 file with one field of view and multiple channels to a multidimensional numpy array.
+def nd2_to_tiff(file: str, channel_order_flip: bool = False, verbose: bool = False) -> np.ndarray:
+    """Converts a single ND2 file with one field of view and multiple channels to a multidimensional numpy array, 
+    ensuring the structure is CYX.
```

Comment:
```suggestion
    """Converts a single ND2 file with one field of view and multiple channels to a multidimensional numpy array, ensuring the structure is CYX.
```

I am following this convention for one line at the top of the docstring because it is enforced by ruff. I would recommend running ruff check in Brieflow/ before filing PR to make sure ruff doesn't complain about anything.

Reply:
Fixed, thanks for the heads up.